ui-spacetimechart/src/stories/split.stories.tsx

```diff
@@ -0,0 +1,185 @@
+import React, { useCallback, useMemo, useState } from 'react';
+
+import '@osrd-project/ui-core/dist/theme.css';
+
+import type { Meta } from '@storybook/react';
+import { clamp, keyBy } from 'lodash';
+
+import { OPERATIONAL_POINTS, PATHS } from './lib/paths';
+import { PathLayer } from '../components/PathLayer';
+import { SpaceTimeChart } from '../components/SpaceTimeChart';
+import type { DrawingFunction, Point } from '../lib/types';
+import { getDiff } from '../utils/vectors';
+import { X_ZOOM_LEVEL, Y_ZOOM_LEVEL, zoom } from './lib/utils';
+import { useDraw } from '../hooks/useCanvas';
+import { AMBIANT_A10 } from '../lib/consts';
+
+const DEFAULT_HEIGHT = 550;
+const COEFFICIENT = 300;
+
+/**
+ * This component renders a colored area where the line only has one track:
+ */
+const FlatStep = ({ position }: { position: number }) => {
+  const drawMonoTrackSpace = useCallback<DrawingFunction>(
+    (ctx, { getSpacePixel, width, height, spaceAxis }) => {
+      const spaceSize = spaceAxis === 'x' ? width : height;
+      const timeSize = spaceAxis === 'x' ? height : width;
+      const fromPixel = clamp(getSpacePixel(position), 0, spaceSize);
+      const toPixel = clamp(getSpacePixel(position, true), 0, spaceSize);
+      const monoLineSize = toPixel - fromPixel;
+      if (!monoLineSize) return;
+
+      ctx.fillStyle = AMBIANT_A10;
+      if (spaceAxis === 'x') {
+        ctx.fillRect(fromPixel, 0, monoLineSize, timeSize);
+      } else {
+        ctx.fillRect(0, fromPixel, timeSize, monoLineSize);
+      }
+    },
+    [position]
+  );
+
+  useDraw('overlay', drawMonoTrackSpace);
+
+  return null;
+};
+
+type WrapperProps = {
+  swapAxis: boolean;
+};
+
+const SplitSpaceTimeChartWrapper = ({ swapAxis }: WrapperProps) => {
+  const [state, setState] = useState<{
+    xOffset: number;
+    yOffset: number;
+    xZoomLevel: number;
+    yZoomLevel: number;
+    panning: null | { initialOffset: Point };
+  }>({
+    xOffset: 0,
+    yOffset: 0,
+    xZoomLevel: X_ZOOM_LEVEL,
+    yZoomLevel: Y_ZOOM_LEVEL,
+    panning: null,
+  });
+
+  // For this story, we split the chart on "City C" and "City E:
+  const splitPoints = useMemo(() => {
+    const operationalPointsDict = keyBy(OPERATIONAL_POINTS, 'id');
+    return [
+      {
+        position: operationalPointsDict['city-c'].position,
+        label: operationalPointsDict['city-c'].label,
+        height: 100,
+      },
```

Comment:
```suggestion
      {
        position: operationalPointsDict['city-c'].position,
        label: operationalPointsDict['city-c'].label,
        height: 100 * state.yZoomLevel,
      },
```

if it's on purpose, you should add a comment

Reply:
It was on purpose, to showcase that it is possible to add both steps with constant (in pixel) heights, and steps with a height that scales with the zoom. I'll add a comment while moving the breakpoints to the story arguments.

Reply:
I just pushed new arguments for the story, including the height of the steps, and a flag to specify if the steps should scale with the zoom or not.